code/modules/projectiles/updated_projectiles/guns/shotguns.dm

```diff
@@ -635,6 +640,90 @@ can cause issues with ammo types getting mixed up during the burst.
 	return TRUE
 
 
+/obj/item/weapon/gun/shotgun/pump/reload_into_chamber(mob/user)
+	if(active_attachable && active_attachable.flags_attach_features & ATTACH_PROJECTILE)
+		make_casing(active_attachable.type_of_casings)
+	else
+		pump_lock = FALSE //fired successfully; unlock the pump
+		current_mag.used_casings++ //The shell was fired successfully. Add it to used.
+		in_chamber = null
+		//Time to move the tube position.
+		if(!current_mag.current_rounds && !in_chamber)
+			update_icon()//No rounds, nothing chambered.
+
+	return TRUE
+
+//***********************************************************
+// Yee Haw it's a cowboy lever action gun!
+
+/obj/item/weapon/gun/shotgun/pump/lever
+	name = "\improper Lever Action rifle"
+	desc = "A 44 magnum lever action rifle with side loading port."
+	icon_state = "mares_leg"
+	item_state = "mares_leg" //With thanks to D4n0w4r https://youtu.be/PV4uWGGb4xM
+	fire_sound = 'sound/weapons/guns/fire/revolver.ogg'//I like how this one sounds.
+	dry_fire_sound = 'sound/weapons/guns/fire/sniper_empty.ogg'
+	reload_sound = 'sound/weapons/guns/interact/mosin_reload.ogg'
+	caliber = ".44 magnum" //codex
+	load_method = SINGLE_CASING //codex
+	max_shells = 10 //codex
+	current_mag = /obj/item/ammo_magazine/internal/shotgun/pump/lever
+	gun_skill_category = GUN_SKILL_RIFLES
+	type_of_casings = "cartridge"
+	pump_sound = 'sound/weapons/guns/interact/mar40_cocked.ogg'//good enough for now.
+	attachable_allowed = list(
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/scope/mini,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/bayonet)
+	attachable_offset = list("muzzle_x" = 50, "muzzle_y" = 21,"rail_x" = 8, "rail_y" = 21, "under_x" = 37, "under_y" = 16, "stock_x" = 20, "stock_y" = 14)
+
+/obj/item/weapon/gun/shotgun/pump/bolt/set_gun_config_values()
+	fire_delay = CONFIG_GET(number/combat_define/med_fire_delay) * 2
+	accuracy_mult = CONFIG_GET(number/combat_define/base_hit_accuracy_mult) + CONFIG_GET(number/combat_define/hmed_hit_accuracy_mult)
+	accuracy_mult_unwielded = CONFIG_GET(number/combat_define/base_hit_accuracy_mult) - CONFIG_GET(number/combat_define/hmed_hit_accuracy_mult)
+	scatter = CONFIG_GET(number/combat_define/low_scatter_value)
+	scatter_unwielded = CONFIG_GET(number/combat_define/max_scatter_value)
+	damage_mult = CONFIG_GET(number/combat_define/base_hit_damage_mult)
+	recoil = CONFIG_GET(number/combat_define/low_recoil_value)
+	recoil_unwielded = CONFIG_GET(number/combat_define/high_recoil_value)
+	pump_delay = CONFIG_GET(number/combat_define/mhigh_fire_delay)
+
+/obj/item/weapon/gun/shotgun/pump/lever/unique_action(mob/user)
+	return work_the_lever(user)
+
+
+/obj/item/weapon/gun/shotgun/pump/proc/work_the_lever(mob/user)	//We can't fire bursts with pumps.
```

Comment:
I'll leave this to Roh to judge, but this seems like a pretty copypasted proc and code duplication is an issue